prime-router/src/main/kotlin/azure/SubmissionFunction.kt

```diff
@@ -82,68 +86,89 @@ class SubmissionFunction(
+                logger.warn(
+                    "Invalid Authorization for user ${claims.userName}:" +
+                        " ${request.httpMethod}:${request.uri.path}." +
+                        " ERR: Claim org is ${claims.organizationNameClaim} but client id is ${sender.organizationName}"
                 )
-                HttpUtilities.okResponse(request, submissions)
+                return HttpUtilities.unauthorizedResponse(request, authorizationFailure)
+            }
+            logger.info(
+                "Authorized request by org ${claims.organizationNameClaim}" +
+                    " to $sender/submissions endpoint via client id ${sender.organizationName}. "
+            )
+
+            val (qSortOrder, qSortColumn, resultsAfterDate, resultsBeforeDate, pageSize, showFailed) =
+                Parameters(request.queryParameters)
+            val sortOrder = try {
+                SubmissionAccess.SortOrder.valueOf(qSortOrder)
             } catch (e: IllegalArgumentException) {
-                HttpUtilities.badRequestResponse(request, e.message ?: "Invalid Request")
+                SubmissionAccess.SortOrder.DESC
             }
+
+            val sortColumn = try {
+                SubmissionAccess.SortColumn.valueOf(qSortColumn)
+            } catch (e: IllegalArgumentException) {
+                SubmissionAccess.SortColumn.CREATED_AT
+            }
+            val submissions = submissionsFacade.findSubmissionsAsJson(
+                sender.organizationName,
+                sortOrder,
+                sortColumn,
+                resultsAfterDate,
+                resultsBeforeDate,
+                pageSize,
+                showFailed
+            )
+            return HttpUtilities.okResponse(request, submissions)
+        } catch (e: IllegalArgumentException) {
+            return HttpUtilities.badRequestResponse(request, e.message ?: "Invalid Request")
         }
     }
 
     /**
      * API endpoint to return history of a single report.
      * The [id] can be a valid UUID or a valid actionId (aka submissionId, to our users)
      */
-    @FunctionName("getReportHistory")
-    fun getReportHistory(
+    @FunctionName("getReportDetailedHistory")
+    fun getReportDetailedHistory(
         @HttpTrigger(
-            name = "getReportHistory",
+            name = "getReportDetailedHistory",
             methods = [HttpMethod.GET],
             authLevel = AuthorizationLevel.ANONYMOUS,
             route = "waters/report/{id}/history"
         ) request: HttpRequestMessage<String?>,
         @BindingName("id") id: String,
     ): HttpResponseMessage {
         try {
-            val claims = oktaAuthentication.authenticate(request)
-                ?: return HttpUtilities.unauthorizedResponse(request, OktaAuthentication.authenticationFailure)
+            // Do authentication
+            val claims = AuthenticationStrategy.authenticate(request)
```

Comment:
And here's the third place its being used (so far!)

Reply:
It does check the claims and do authorization based on them, below on line 187 in the new code:
`  if (!submissionsFacade.checkSenderAccessAuthorization(action, claims)) {`
(its so much later because the Report Details API query _has no organization name in it,_ so first we have to do a bunch of database queries to figure out who "owns" the report)

I'd like to have a single uniform way that checks claims, but that will require a single standard way of managing them, so we're another round of refactoring away from that.   So, for right now, I'm pleased that I've got a single authentication(...) method, but we do not yet have a single authorization(...) method.